File: 卡码/代码随想录/JS/二叉树/02-二叉树属性/LeetCode_0559.js
//559. N 叉树的最大深度
function Node(val,children) {
  this.val = val;
  this.children = children;
};

const maxDepth = function(root) {
  if(!root) return 0;
  let depth=0;
  for(let node of root.children){
    depth=Math.max(depth,maxDepth(node));
  }
  return 1+depth;
};

File: 卡码/代码随想录/Python/二叉树/02-二叉树属性/LeetCode_0559.py
#559. N 叉树的最大深度
from typing import *

class Node:
  def __init__(self, val=None, children=None):
    self.val = val
    self.children = children

class Solution:
  def maxDepth(self, root: 'Node') -> int:
    if not root:
      return 0
    depth=0
    for node in root.children:
      depth=max(depth,self.maxDepth(node))
    return 1+depth
